<?xml version="1.0" encoding="utf-8"?>
<robot name="mopa" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <material name="dark_grey">
    <color rgba="0.2 0.2 0.2 1.0"/>
  </material>  

  <link name="base_box"/>
  <joint name="base_link_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="base_box"/>
    <child link="ridgeback_dummy"/>
  </joint>
  <link name="ridgeback_dummy">
    <inertial>
      <mass value="100"/>
      <origin rpy="0 0 0" xyz="0 0 0.075"/>
      <inertia ixx="4.27" ixy="0" ixz="0" iyy="5.52" iyz="0" izz="9.42"/>
    </inertial>
    <visual>
      <origin xyz="0 0 0.075" rpy="0 0 0" />
      <geometry>
        <box size="0.8 0.7 0.15"/>
      </geometry>
      <material name="dark_grey"/>
    </visual>
    <collision>
      <origin xyz="0 0 0.075" rpy="0 0 0" />
      <geometry>
        <box size="0.8 0.7 0.15"/>
      </geometry>
    </collision>
  </link>

  <xacro:include filename="$(find franka_description)/robots/panda_arm.xacro"/>
  <xacro:include filename="$(find franka_description)/robots/hand.xacro"/>
  <xacro:panda_arm connected_to="ridgeback_dummy" xyz="0.2 0 0.45" rpy="0 0 0" />
  <xacro:hand ns="panda" rpy="0 0 ${-pi/4}" connected_to="panda_link8"/>
  
  <!-- Add EE frame -->
  <link name="panda_default_EE">
    <inertial>
      <mass value="0.01"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="3.0e-07" ixy="0.0" ixz="0.0" iyy="3.0e-07" iyz="0.0" izz="3.0e-07"/>
    </inertial>
  </link>
  <joint name="joint_link8_default_ee" type="fixed">
    <parent link="panda_link8"/>
    <child link="panda_default_EE"/>
    <origin rpy="0.000 0.000 ${-pi/4}" xyz="0.000 0.000 0.103"/>
  </joint>

</robot>
